portal-impl/src/com/liferay/portal/service/impl/WorkflowDefinitionLinkLocalServiceImpl.java

```diff
@@ -43,12 +44,26 @@ public WorkflowDefinitionLink addWorkflowDefinitionLink(
 			int workflowDefinitionVersion)
 		throws PortalException {
 
-		User user = _userPersistence.findByPrimaryKey(userId);
-
 		long workflowDefinitionLinkId = counterLocalService.increment();
```

Comment:
Can you move this line to just before the `workflowDefinitionLinkPersistence.create(workflowDefinitionLinkId)` logic like it was before?

Reply:
We don't use the user variable for anything before line 71, and since we have a possible exception case before that, it's more efficient and performant to initialize it after ensuring that the code will need to be executed/after checking for the exception.